iothub/device/samples/PnpDeviceSamples/TemperatureController/ComponentTemperatureControllerSampleOld.cs

```diff
@@ -86,39 +87,40 @@ public async Task PerformOperationsAsync(CancellationToken cancellationToken)
             };
 
             await _deviceClient.SendEventAsync(message, cancellationToken);
-            _logger.LogDebug($"Telemetry: Sent - {JsonConvert.SerializeObject(telemtry)} in KB.");
+            _logger.LogDebug($"Telemetry: Sent - {System.Text.Json.JsonSerializer.Serialize(telemetry)}.");
 
             // Subscribe and respond to event for writable property "humidityRange" under component "thermostat1".
             await _deviceClient.SetDesiredPropertyUpdateCallbackAsync(async (desired, userContext) =>
             {
                 string propertyName = "humidityRange";
-                if (!desired.Contains(Thermostat1) || !((JObject)desired[Thermostat1]).ContainsKey(propertyName))
+                if (!desired.Contains(Thermostat1) || !((JObject)desired[Thermostat1]).TryGetValue(propertyName, out JToken humidityRangeToken))
                 {
                     _logger.LogDebug($"Property: Update - Received a property update which is not implemented.\n{desired.ToJson()}");
                     return;
                 }
 
-                HumidityRange targetHumidityRange = ((JObject)desired[Thermostat1]).Value<HumidityRange>(propertyName);
+                HumidityRange targetHumidityRange = System.Text.Json.JsonSerializer.Deserialize<HumidityRange>(humidityRangeToken.ToString());
 
-                var propertyPatch = new TwinCollection();
-                var componentPatch = new TwinCollection()
+                var propertyPatch = new Dictionary<string, object>();
+                var componentPatch = new Dictionary<string, object>
```

Comment:
Why remove the TwinCollection? Did it fail? This is in our docs.

Reply:
Using TwinCollection as the type to hold the property value will work as expected as long as the "value" is either a primitive type or a newtonsoft.json annotated type. This is because TwinCollection will use newtonsoft to serialize the object.
However, if the "value" is type that uses a different serializer, but we initialize it to a TwinCollection directly, the resultant property patch will no longer have the correct json property keys. For this reason, we need to serialize the object based on custom serializer and then feed the serialized string to TwinCollection. This serialized string can be internally represented as a JObject (or any other newtonsoft type) as required, it won't have any additional negative impact.

Reply:
This results in:
```json
{{
  "thermostat1": {
    "__t": "c",
    "humidityRange": {
      "value": {
        "MaxTemperature": 50.0,
        "MinTemperature": 20.0
      },
      "ac": 200,
      "av": 8,
      "ad": "The operation completed successfully."
    }
  }
}}
```
instead of
```json
{{
  "thermostat1": {
    "__t": "c",
    "humidityRange": {
      "value": {
        "maxHumidity": 50,
        "minHumidity": 20
      },
      "ac": 200,
      "av": 9,
      "ad": "The operation completed successfully."
    }
  }
}}
```